borderless/explainer.md

```diff
@@ -0,0 +1,512 @@
+    - [2. Enabling draggable regions](#2-enabling-draggable-regions)
+    - [3. CSS display-mode media query for borderless](#3-css-display-mode-media-query-for-borderless)
+    - [4. Leveraging Window Management permission for borderless](#4-leveraging-window-management-permission-for-borderless)
+      - [Enabling Window Management](#enabling-window-management)
+      - [Disabling Window Management](#disabling-window-management)
+    - [5. Isolated Web App check](#5-isolated-web-app-check)
+  - [Related features](#related-features)
+    - [Displaying app’s origin in the app settings UI](#displaying-apps-origin-in-the-app-settings-ui)
+    - [Privacy indicators](#privacy-indicators)
+  - [Other notable details](#other-notable-details)
+    - [“Settings and more” three-dot button](#settings-and-more-three-dot-button)
+  - [Demo](#demo)
+    - [manifest.json](#manifestjson)
+    - [index.html](#indexhtml)
+    - [style.css](#stylecss)
+  - [Considered Alternatives](#considered-alternatives)
+    - [Having a separate permission for borderless](#having-a-separate-permission-for-borderless)
+    - [WCO extension](#wco-extension)
+      - [Reasonings for discarding](#reasonings-for-discarding)
+    - [New JS APIs](#new-js-apis)
+      - [Reasonings for discarding](#reasonings-for-discarding-1)
+  - [Security Considerations](#security-considerations)
+    - [Spoofing risks](#spoofing-risks)
+    - [Out-of-scope Navigation](#out-of-scope-navigation)
+    - [Opening a popup from a borderless PWA](#opening-a-popup-from-a-borderless-pwa)
+    - [Iframes](#iframes)
+
+## Introduction
+
+[VDI](https://www.softwaretestinghelp.com/best-vdi-software/#What_Is_Virtual_Desktop_Infrastructure)
+(Virtual Desktop Infrastructure) providers attempt to mirror remote applications
+to the local desktop. For example, the VDI provider's cloud is running a Text
+Processor (a desktop application window), which is to be presented on the client
+desktop as if the application was running locally.
+
+Installed web apps define their display mode in the manifest.json using display
+and
+[display_override](https://github.com/WICG/display-override/blob/main/explainer.md)
+fields. Currently
+[supported values for display modes](<https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/public/mojom/manifest/display_mode.mojom?q=-file:(test%7CDebug%7C%5C.xtb$)%20case:yes%20third_party%2Fblink%2Fpublic%2Fmojom%2Fmanifest%2Fdisplay_mode.mojom&ss=chromium>)
+are `window-controls-overlay`, `tabbed`, `minimal-ui`, `standalone`, `browser`
+and `fullscreen`. See
+[Manifest/display_override#values](https://developer.mozilla.org/en-US/docs/Web/Manifest/display_override#values)
+for more. From the before-mentioned options, `window-controls-overlay` has some
+same qualities to what we are trying to achieve. However, it is still not
+offering enough native-like experience for some of the use-cases and thus this
```

Comment:
I'm confused here. You say you need borderless for a "native-like experience", however, isn't the most "native-like experience", by definition, the `standalone` window mode? That allows the OS to draw its standard title bar on the window however it likes, making the window look exactly like native windows on the host OS.

WCO moves away from the native-like experience but still allows for the OS to draw its native control buttons.

Borderless moves completely away from the native-like experience, allowing the app to draw its own buttons (or not). Traditionally, borderless has been used for apps like [Steam](https://cdn.yurishwedoff.com/will_steam_ever_update_its_ui.jpg) to render a totally customized UI, the exact opposite of native.

I realize you are proposing this for a somewhat different use case, which is that a VDI app on a remote system will draw its own "native" UI, and we do not want the host OS to draw a second duplicate UI. However, let's be clear that, for example, on ChromeOS, the remote Windows UI is not "native"; it is foreign to the system, and that is exactly why you need a new API to draw it.

I think it would be good to break this into three distinct use cases as I see it, and explore each case and whether it is an explicit goal or not:

1. Apps that want to fully customize their title bar to still have the same functionality, but with their own style, like Steam.
2. Apps that want to completely remove the title bar and provide no controls (which can be seen sometimes in native apps for dialogs like splash screens).
3. The VDI case, where the remote app is drawing a remote-OS-native title bar and we want to avoid drawing a host-native title bar over the top of it or above it.

Reply:
I rewrote the intro now, PTAL :) Also please resolve this if you think it's addressed.

Reply:
nit: Change the link to Steam to just `https://store.steampowered.com/` instead of the random screenshot I gave you (it's a random thing I found on Google images). Even though that doesn't have a screenshot directly in the page, it's an authoritative link which people can use to find images if they need to.

Otherwise good.